src/test/java/com/example/gradecalculator/service/UserServiceTest.java

```diff
@@ -296,4 +300,54 @@ public void testGetAuthenticatedUser_UserNotFound() {
         // Act & Assert
         assertThrows(IllegalArgumentException.class, () -> userService.getAuthenticatedUser(authentication));
     }
+
+    @Test
+    public void deleteUserAndSubjects_UserExistsAndDeletedSuccessfully() {
+        // Arrange
+        Long userId = 1L;
+        User user = new User();
+        user.setId(userId);
+
+        when(userRepository.findById(userId)).thenReturn(Optional.of(user));
+        doNothing().when(userSubjectRepository).deleteAllByUserId(userId);
+        doNothing().when(userRepository).delete(user);
+
+        // Act
+        boolean result = userService.deleteUserAndSubjects(userId);
+
+        // Assert
+        assertTrue(result);
+        verify(userSubjectRepository, times(1)).deleteAllByUserId(userId);
+        verify(userRepository, times(1)).delete(user);
+    }
+
+    @Test
+    public void deleteUserAndSubjects_UserDoesNotExist() {
+        // Arrange
+        Long userId = 1L;
+        when(userRepository.findById(userId)).thenReturn(Optional.empty());
+
+        // Act
+        boolean result = userService.deleteUserAndSubjects(userId);
+
+        // Assert
+        assertFalse(result);
+        verify(userSubjectRepository, never()).deleteAllByUserId(anyLong());
+        verify(userRepository, never()).delete(any(User.class));
+    }
+
+    @Test
+    public void deleteUserAndSubjects_UserExistsButDeleteFails() {
```

Comment:
seeing this test, wouldn't it be better if in the case of a repository exception the expected outcome if the service method was false and no exception was thrown?